pandas/core/dtypes/cast.py

```diff
@@ -577,6 +577,9 @@ def maybe_promote(dtype, fill_value=np.nan):
             dtype = np.dtype(np.object_)
         elif is_integer(fill_value) or (is_float(fill_value) and not isna(fill_value)):
             dtype = np.dtype(np.object_)
+        elif is_valid_nat_for_dtype(fill_value, dtype):
+            # e.g. pd.NA, which is not accepted by Timestamp constructor
+            fill_value = np.datetime64("NaT", "ns")
```

Comment:
But pd.NA should not be an allowed fill_value for datetime dtypes?

Reply:
i thought (half of) the whole idea of pd.NA was that it would be a valid NA value for every dtype.

(the other half being kleene logic)

Reply:
I *don't* think we support `pd.NA` atm in datetimelikes (agree eventually we should but we should do this deliberately). However doesn't `is_valid_nat_for_dtype` distinguish this explicitly?

Reply:
> i thought (half of) the whole idea of pd.NA was that it would be a valid NA value for every type.

In the yes, for sure. But at the moment, we don't use pd.NA for datetimelikes, and also don't properly support it in operations with datetimelikes. 

So therefore I am wondering if we should allow it here.

(now, it seems that it already *does* work on master as well, though)

Reply:
> However doesn't is_valid_nat_for_dtype distinguish this explicitly?

is_valid_nat_for_dtype considers pd.NA valid for all dtypes, bc that was my understanding of the intent of pd.NA.